docker/daemon.go

```diff
@@ -79,14 +91,37 @@ func migrateKey() (err error) {
 	return nil
 }
 
-func mainDaemon() {
-	if utils.ExperimentalBuild() {
-		logrus.Warn("Running experimental build")
+// DaemonCli represents the daemon CLI.
+type DaemonCli struct {
+	*daemon.Config
+	registryOptions *registry.Options
+}
+
+// CmdDaemon is the daemon command, called the raw arguments after `docker daemon`.
+func (cli *DaemonCli) CmdDaemon(args ...string) error {
+	if *flDaemon {
+		// allow `docker -D -d` and `docker -d -D`
+		logrus.Warn("please use 'docker daemon' instead.")
+	} else if !(commonFlags.FlagSet).IsEmpty() {
+		// deny `docker -D daemon`
+		logrus.Fatal("invalid global flags")
```

Comment:
if calling Visit() will cause a nicer error message then probably. Don't understand the code well enough yet to know the right code solution.